pkgs/development/cuda-modules/buildRedist/buildRedistHook.bash

```diff
@@ -119,27 +125,76 @@ checkCudaFhsRefs() {
 }
 
 checkCudaNonEmptyOutputs() {
-  local output
+  local outputName
   local dirs
-  local -a failingOutputs=()
-
-  for output in $(getAllOutputNames); do
-    [[ ${!output:?} == "out" || ${!output:?} == "${!outputDev:?}" ]] && continue
-    dirs="$(find "${!output:?}" -mindepth 1 -maxdepth 1)" || true
-    if [[ -z $dirs || $dirs == "${!output:?}/nix-support" ]]; then
-      failingOutputs+=("$output")
+  local -a failingOutputNames=()
+
+  for outputName in $(getAllOutputNames); do
+    # NOTE: Reminder that outputDev is the name of the dev output, so we compare it as-is against outputName rather
+    # than using !outputDev, which would give us the path to the dev output.
+    [[ ${outputName:?} == "out" || ${outputName:?} == "${outputDev:?}" ]] && continue
+    dirs="$(find "${!outputName:?}" -mindepth 1 -maxdepth 1)" || true
+    if [[ -z $dirs || $dirs == "${!outputName:?}/nix-support" ]]; then
+      failingOutputNames+=("${outputName:?}")
     fi
   done
 
-  if ((${#failingOutputs[@]})); then
-    nixErrorLog "detected empty (excluding nix-support) outputs: ${failingOutputs[*]}"
+  if ((${#failingOutputNames[@]})); then
+    nixErrorLog "detected empty (excluding nix-support) outputs: ${failingOutputNames[*]}"
     nixErrorLog "this typically indicates a failure in packaging or moveToOutput ordering"
     exit 1
   fi
 
   return 0
 }
 
+checkCudaHasStubsIffIncludeRemoveStubsFromRunpathHook() {
+  local outputName
+  local -i hasStubs
+  local -i hasRemoveStubsFromRunpathHook
+  local -a outputNamesWronglyExcludingHook=()
+  local -a outputNamesWronglyIncludingHook=()
+
+  for outputName in $(getAllOutputNames); do
+    hasStubs=0
+    if find "${!outputName:?}" -mindepth 1 -type d -name stubs -print -quit | grep --silent .; then
```

Comment:
NOTE: false for `${drv.stubs}/lib/libfoo.so`, but well right now don't have any left

Reply:
Updated in 4944bbd7ae604d82e48c5f894abe89e67b5d20f5 to check if the output name is `stubs`.